ansible/files/auth_index.json

```diff
@@ -2,7 +2,7 @@
   "_id":"_design/subjects",
   "views": {
     "identities": {
-      "map": "function (doc) {\n  if(doc.uuid && doc.key) {\n    var v = {namespace: doc.subject, uuid: doc.uuid, key: doc.key};\n    emit([doc.subject], v);\n    emit([doc.uuid, doc.key], v);\n  }\n  if(doc.namespaces) {\n    doc.namespaces.forEach(function(namespace) {\n      var v = {namespace: namespace.name, uuid: namespace.uuid, key: namespace.key};\n      emit([namespace.name], v);\n      emit([namespace.uuid, namespace.key], v);\n    });\n  }\n}"
+      "map": "function (doc) {\n  if(doc.uuid && doc.key) {\n    var v = {namespace: doc.subject, uuid: doc.uuid, key: doc.key};\n    emit([doc.subject], v);\n    emit([doc.uuid, doc.key], v);\n  }\n  if(doc.namespaces) {\n    var subject = doc.subject;\n    doc.namespaces.forEach(function(namespace) {\n      var v = {namespace: namespace.name, uuid: namespace.uuid, key: namespace.key};\n      emit([namespace.name], v);\n      emit([namespace.uuid, namespace.key], v);\n      emit([subject], v);\n    });\n  }\n}"
     }
```

Comment:
The function defines here 
{
   "identities": {
       "map": "function (doc) {
          if(doc.uuid && doc.key) {
            var v = {namespace: doc.subject, uuid: doc.uuid, key: doc.key};  
            emit([doc.subject], v);    
            emit([doc.uuid, doc.key], v);
          } 
          if(doc.namespaces) { 
            `var subject = doc.subject; `
            doc.namespaces.forEach(function(namespace) { 
            var v = {namespace: namespace.name, uuid: namespace.uuid, key: namespace.key};
            emit([namespace.name], v);
            emit([namespace.uuid, namespace.key], v);
            `emit([subject], v);`
            });
             }
            }"
   }
}
So, i just added two lines in `if(doc.namespaces)` condition. Why i add `this two lines`, the reason is that: the client certificate will pass `CN` value to controller(the `CN` value is username, it is `subject concept in openwhisk),  so it can get the `Identity` by `username` from couchdb, so should modify above view's  function